### ISSUE TITLE
Lyric spacing is not always consistent

### ISSUE THREAD

**Issue by mwinckler** [mwinckler/LyricsAnimator#4]
In songs with inconsistent lyric lengths (or even inconsistent lyric line lengths), the spacing between lyrics as they scroll is sometimes inconsistent. It should be possible to calculate the average space required based on the times in the file and fudge the start frame of lyrics to make the spaces more visually pleasing.

An example troublesome song in this regard is Hallelujah, Praise Jehovah.

**Comment by mwinckler** [comment]
Another cause of this problem is final verses which have significant _ritardando._ When the speed changes within the verse, it makes that the "slowest" verse which is taken into consideration on the others. We could get around this by subtly changing the scroll speed within a verse itself via easing - if no other verses are on screen, the difference should not be noticeable.

Alternatively, put all verses into one long text block and provide "hinting" times for when the scrolling should reach given points. Then hints could be used around tempo changes with subtle easing to adjust the animation speed.

**Comment by mwinckler** [comment]
Proposed config format:

* Lines prefixed with a timestamp in square brackets should arrive at the ideal reading height at that timestamp.
* Text inside curly braces will be displayed as the "verse number" at the right of the screen and remain in place until the next curly brace text arrives.

Example:

```
[00:00:12] {verse 1} Low in the grave he lay— Jesus, my Savior,
waiting the coming day— Jesus, my Lord.
[00:00:35] Up from the grave he arose, with a mighty triumph o'er his foes.
He arose a victor from the dark domain, and he lives forever with his saints to reign.
He arose! He arose! Hallelujah! Christ arose!
[00:01:03] {verse 2} Vainly they watch his bed— Jesus, my Savior;
vainly they seal the dead— Jesus, my Lord.
[00:01:27] Up from the grave he arose, with a mighty triumph o'er his foes.
He arose a victor from the dark domain, and he lives forever with his saints to reign.
He arose! He arose! Hallelujah! Christ arose!
[00:01:56] {verse 3} Death cannot keep his prey— Jesus, my Savior;
he tore the bars away— Jesus, my Lord.
[00:02:19] Up from the grave he arose, with a mighty triumph o'er his foes.
He arose a victor from the dark domain, and he lives forever with his saints to reign.
He arose! He arose! Hallelujah! Christ arose!
[00:02:51]
```

**Comment by mwinckler** [comment]
Fixed in daa75d7dc81a9a850fc1bc04742f802206bfdb24.
